packages/@aws-cdk/aws-cloudwatch/lib/metric.ts

```diff
@@ -216,10 +236,17 @@ export class Metric implements IMetric {
     if (periodSec !== 1 && periodSec !== 5 && periodSec !== 10 && periodSec !== 30 && periodSec % 60 !== 0) {
       throw new Error(`'period' must be 1, 5, 10, 30, or a multiple of 60 seconds, received ${periodSec}`);
     }
+    if (props.dimensions && props.dimensionsMap) {
```

Comment:
I feel like you can safely skip this check, and simply ignore `props. dimensions` if `props. dimensionsMap` was provided.

If you decide to go with my suggestion of populating `this.dimensions` from `props.dimensionsMap`, removing this check might be required to get the `with` method working in its current shape.